app/src/main/kotlin/com/backbase/golden_sample_app/presentation/header/TopBarConfigurationProvider.kt

```diff
@@ -0,0 +1,29 @@
+package com.backbase.golden_sample_app.presentation.header
+
+import com.backbase.android.business.journey.common.user.UserRepository
+import com.backbase.android.design.header.AvatarConfiguration
+import com.backbase.android.design.header.TopBarConfiguration
+import com.backbase.android.plugins.storage.StorageComponent
+import com.backbase.golden_sample_app.router.getUsernameOrBlank
+
+class TopBarConfigurationProvider(
+    private val storageComponent: StorageComponent,
+    private val userRepository: UserRepository,
+) {
+
+    fun config() = TopBarConfiguration {
+        val userFullName = storageComponent.getUsernameOrBlank()
+
+        avatar = AvatarConfiguration { initials = userFullName.toNameInitials() }
+        title = userFullName
```

Comment:
This won't work for first time logging in and changing user

Reply:
I would change it dynamically like you set text/images for any view. Same would go later on when there is the ability to set an avatar picture. We won't want the user to close the app first before you see the result

Reply:
UserProfile Usecase has [UserPersonalInformation](https://backbase.io/developers/android/capabilities/user-profile/5.1.0/user-profile-journey/com.backbase.android.identity.journey.userprofile.models/-user-personal-information/index.html) which contains the user name.

Officially UseCases are not meant to be reused as public api, but Identity does seem to treat it like so. My suggestion was we create a repository that can retrieve the data (via a passed on UseCase). We can then either use it in the Activity, but way cleaner would of course be to use this in a ViewModel. That last bit you don't need to do, we can update that.